src/middleware/serverMiddlewareLoader.middleware.ts

```diff
@@ -27,6 +32,20 @@ export default class ServerMiddlewareLoader {
     app.use(bodyParser.urlencoded({ extended: true }))
     app.use(cookieParser())
     app.use(createLoggerMiddleware(APP_NAME))
+    app.use(helmet.hidePoweredBy())
+    app.use(
+      helmet.contentSecurityPolicy({
+        directives: {
+          defaultSrc: ['\'self\''],
+          scriptSrc: ['\'self\'', 'code.jquery.com', this.CDN_HOST, '\'unsafe-inline\'',
```

Comment:
why is unsafe-inline needed?

Reply:
We have some inline scripts on the page, I believe PIWIK related

Reply:
`    <script>document.body.className = ((document.body.className) ? document.body.className + ' js-enabled' : 'js-enabled');</script>`

`    <script>document.body.className = ((document.body.className) ? document.body.className + ' js-enabled' : 'js-enabled');</script>`

` <script type="application/javascript">
    (function() {
      bindPiwikListener('dissolutions', 'http://piwiki', 999);
    }());
  </script>`

Turns out they are govuk related as well

Reply:
Taken from https://content-security-policy.com/unsafe-inline/
> The unsafe-inline keyword annuls most of the security benefits that Content-Security-Policy provide.

Instead, check out the `What should you use instead of unsafe-inline?` section in the above link

Reply:
jquery min also complains.

Reply:
Yup we should use a `nonce` to ensure that the inline scripts are safe. To define a `nonce` value:
1. Create a nonce variable using the uuid library we currently use
2. Bind it to `res.locals` so that it is accessible in nunjucks
```
const nonce: string = uuidV4()

app.use((req, res, next) => {
  res.locals.nonce = nonce
})
```
3. Update these script tags to use the nonce
```
<script nonce={{ nonce | dump | safe }}>
  document.body.className = ((document.body.className) ? document.body.className + ' js-enabled' : 'js-enabled');
</script>
```
4. Update the CSP to whitelist these scripts that utilise the `nonce` value:
```
`'nonce-${nonce}'`
```

Reply:
Also consider using backticks through to avoid having to escape single quotes with backslashes